src/search.js

```diff
@@ -1,23 +1,25 @@
-import {assoc, concat, filter, has, isEmpty, map, pipe, prop, sortBy} from "ramda"
+import {ascend, assoc, concat, descend, has, isEmpty, map, prop, sortBy, sortWith} from "ramda"
 import * as diacritics from 'diacritics'
 
 
 export function findParametersAndVariables(parameters, variables, query) {
-  function matchesQuery(parameterOrVariable) {
-    if (isEmpty(normalizedQuery)) {
-      return true
-    }
-    return parameterOrVariable.name.includes(normalizedQuery) ||
-      parameterOrVariable.searchIndex.includes(normalizedQuery)
+  function indexInString(substring, string) {
+    const index = string.indexOf(substring)
+    return index === -1 ? Number.MAX_SAFE_INTEGER : index
   }
   const normalizedQuery = diacritics.remove(query.toLowerCase())
-  return pipe(
-    filter(matchesQuery),
-    sortBy(prop('name')),
-  )(concat(
+  let parametersAndVariables = concat(
     map(assoc('type', 'parameter'), parameters),
     map(assoc('type', 'variable'), variables),
-  ))
+  )
+  const sortFunction = isEmpty(normalizedQuery)
+    ? sortBy(prop('name'))
+    : sortWith([
+      ascend(item => indexInString(normalizedQuery, item.name)),
+      descend(item => item.searchIndex.includes(normalizedQuery)),
```

Comment:
OK.

After having re-read the whole thing, it appears that the `searchIndex` property is the description without diacritics. It's not added by the search function because it's bad for performance as the same computation would be done on every search. It's done once at application start, when it fetches the API data from `parameters` and `variables` endpoints.

Perhaps it should be named `normalizedDescription`, but parameters have labels, so `normalizedDescriptionOrLabel` should be more exact, but ugly.